torch/nn/modules/sparse.py

```diff
@@ -198,7 +199,8 @@ def forward(self, input, offsets=None):
                                  " fixed length sequences. However, found "
                                  "offsets of type {}".format(type(offsets)))
             else:
-                offsets = input.data.new(input.size(0)).fill_(input.size(1))
+                offsets = Variable(torch.arange(0, input.numel(), input.size(1)).long())
```

Comment:
This won't work with CUDA inputs. This would be better:
```python
offsets = Variable(torch.arange(0, input.numel(), input.size(1), out=input.data.new().long()))
```

Can you also modify the test so that it checks CUDA implementation too? It'd be the best to refactor this function to take an additional `cast` arg (like [in here](https://github.com/ajfisch/pytorch/blob/master/test/test_torch.py?utf8=%E2%9C%93#L501-L520)), and then add CPU and CUDA tests that call it with different lambdas